src/Language/Haskell/Liquid/Parse.hs

```diff
@@ -750,13 +749,18 @@ tyBodyP ty
           outTy (RFun _ _ t _) = Just t
           outTy _              = Nothing
 
+upperIdP' :: Parser Symbol
+upperIdP' = try $ symbol <$> condIdP alphanums (isUpper . head)
+  where
+    alphanums = ['A' .. 'Z'] ++ ['a'..'z'] ++ ['0'..'9'] ++ "'"
```

Comment:
We should also allow unicodes and generally follow haskell's grammar https://www.haskell.org/onlinereport/syntax-iso.html

But lets merge this (once rebased and Travis accepts it) and @Kyly will fix it later.